spiped/dispatch.c

```diff
@@ -80,7 +82,7 @@ doaccept(struct accept_state * A)
 	int rc = 0;
 
 	/* If we can, accept a new connection. */
-	if ((A->nconn < A->nconn_max) && (A->accept_cookie == NULL)) {
+	if ((A->nconn < A->nconn_max) && (A->accept_cookie == NULL) && !(A->shutdown)) {
```

Comment:
> doaccept seems to prepare for an incoming connection in advance, rather than choosing whether to accept a connection which is happening right now. I would be tempted to call it "prepare_for_future_connection".

doaccept calls network_accept, which is a non-blocking version of accept.  The name is in analogy with the behaviour of accept, which waits until a connection arrives and then accepts it.